src/osd/PrimaryLogPG.cc

```diff
@@ -5790,9 +5802,14 @@ int PrimaryLogPG::do_osd_ops(OpContext *ctx, vector<OSDOp>& ops)
       }
       ++ctx->num_write;
       { // zero
-	result = check_offset_and_length(op.extent.offset, op.extent.length, cct->_conf->osd_max_object_size);
-	if (result < 0)
+	result = check_offset_and_length(op.extent.offset, op.extent.length,
+          osd_max_object_size);
+	if (result < 0) {
+          derr << __func__
+            << " osd_max_object_size >= 4GB; BlueStore has hard limit of 4GB."
+            << dendl;
```

Comment:
How about moving the error logging into check_offset_and_length() instead of duplicating it everywhere that function is called?